Car basic features added

Caaki opened the pull request:
Added basic crude car features with an added HttpResponse class for better and easier testing and implementation on the front end.

sergienkoyura commented on src/main/java/com/savicsoft/carpooling/car/model/entity/Car.java:
```diff
@@ -19,7 +26,19 @@ public class Car {
+    private int year;
+    @Enumerated(EnumType.STRING)
+    private FuelType fuelType;
+    private double fuelEfficiency;
```
Do we need fuel type and efficiency params? As for me, it is just extra information but lets discuss

sergienkoyura reviewed (commented):
What about DTO? I mean, we can add a VIN code to verify the car for example, and show up all the params in DTO except VIN (or fuel type - efficiency).

Caaki commented:
I added the service to work with the new DTO objects and mappers, I removed the id so everything will go using UUID and fuel efficiency attribute from the DTO everything else is the same.

And now the car has a new attribute for the picture url.

Caaki commented on src/main/java/com/savicsoft/carpooling/car/dto/mapper/CarDTOMapper.java:
```diff
@@ -0,0 +1,18 @@
+import org.springframework.context.annotation.Bean;
+import org.springframework.stereotype.Service;
+
+public class CarDTOMapper {
```
Want to go on a call so we can edit it together :)

Caaki commented on src/main/java/com/savicsoft/carpooling/car/dto/mapper/CarDTOMapper.java:
I changed the DTOMapper to use Mapper instead of the BeanUtils,
And implemented the same changes to the service.
